### PR TITLE
enable/disable debug message

### PR DESCRIPTION
**Pull request by ochinchina**

### CODE REVIEW

File: reaper.go [hunk 1]
```diff
@@ -14,6 +14,8 @@ type Config struct {
 	DisablePid1Check bool
 }
 
+var debug bool = false
```
Comment (ramr) [inline reaper.go:17]: Makes more sense to add it to the config structure above. That way we don't need an on/off method - the config variable will drive the debugging messages.

File: reaper.go [hunk 2]
```diff
@@ -47,7 +49,9 @@ func reapChildren(config Config) {
 
 	for {
 		var sig = <-notifications
-		fmt.Printf(" - Received signal %v\n", sig)
+		if debug {
```
Comment (ramr) [inline reaper.go:52]: use `config.Debug` -- see comment above about adding to config struct.

File: reaper.go [hunk 3]
```diff
@@ -64,14 +68,25 @@ func reapChildren(config Config) {
 				break
 			}
 
-			fmt.Printf(" - Grim reaper cleanup: pid=%d, wstatus=%+v\n",
-				pid, wstatus)
+			if debug {
```
Comment (ramr) [inline reaper.go:71]: use `config.Debug` -- see comment above about adding to config struct.

File: reaper.go [hunk 4]
```diff
@@ -64,14 +68,25 @@ func reapChildren(config Config) {
...
+				fmt.Printf(" - Grim reaper cleanup: pid=%d, wstatus=%+v\n",
+					pid, wstatus)
+			}
 
 		}
 	}
 
 } /*   End of function  reapChildren.  */
 
+func EnableDebug() {
```
Comment (ramr) [inline reaper.go:82]: not needed anymore.

File: reaper.go [hunk 5]
```diff
@@ -64,14 +68,25 @@ func reapChildren(config Config) {
...
+	debug = true
+}
+
+func DisableDebug() {
```
Comment (ramr) [inline reaper.go:86]: same here - not needed if you add to the config struct.

File: reaper.go [hunk 6]
```diff
@@ -109,7 +124,9 @@ func Start(config Config) {
 	if !config.DisablePid1Check {
 		mypid := os.Getpid()
 		if 1 != mypid {
-			fmt.Printf(" - Grim reaper disabled, pid not 1\n")
+			if debug {
```
Comment (ramr) [inline reaper.go:127]: use `config.Debug` -- see comment above about adding to config struct.

**Review by ramr — CHANGES_REQUESTED** [review]

**Comment by ramr** [comment]
@ochinchina  thanks for the PR. My apologies for the delay looking at this - just got a chance. Added some review comments and we can simplify this with just a config option. Thanks.

File: reaper.go (hunk 1 above)
Comment (ochinchina) [reply reaper.go:17]: That is good idea to expose debug flag in the Config data structure

**Comment by sergeyklay** [comment]
@ochinchina Could you please apply changes described in the comments

**Pull request closed without merging**